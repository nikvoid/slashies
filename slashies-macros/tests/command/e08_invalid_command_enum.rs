use slashies_macros::Command;

/// An command with an invalid subcommand structure
#[derive(Command)]
#[name = "BadCommand"]
enum BadCommand {
    SubCommand { named: u64, fields: u64 },
}

/// An command with an invalid subcommand structure
#[derive(Command)]
#[name = "BadCommand"]
enum BadCommand2 {
    SubCommand(u64, u64),
}

/// An command with an invalid subcommand structure
#[derive(Command)]
#[name = "BadCommand"]
enum BadCommand3 {
    SubCommand,
}

fn main() {}
