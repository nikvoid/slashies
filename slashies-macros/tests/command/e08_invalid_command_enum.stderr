error: Subcommand variants for a Command enum must have unnamed fields
 --> tests/command/e08_invalid_command_enum.rs:7:5
  |
7 |     SubCommand { named: u64, fields: u64 },
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Variants of a Command enum must be a tuple of length 1, containing only the subcommand
  --> tests/command/e08_invalid_command_enum.rs:14:16
   |
14 |     SubCommand(u64, u64),
   |                ^^^^^^^^

error: Subcommand variants for a Command enum must have unnamed fields
  --> tests/command/e08_invalid_command_enum.rs:21:5
   |
21 |     SubCommand,
   |     ^^^^^^^^^^
